include/odp/api/spec/traffic_mngr.h

```diff
@@ -584,6 +614,41 @@ void odp_tm_egress_init(odp_tm_egress_t *egress);
 int odp_tm_capabilities(odp_tm_capabilities_t capabilities[],
 			uint32_t              capabilities_size);
 
+/** Query All TM Capabilities specific to an egress
+ *
+ * The odp_tm_egress_capabilities() function can be used to obtain the complete
+ * set of TM limits supported by this implementation for a given egress.
+ * The reason that this  returns a SET of capabilities and not just one, is
+ * because it is expected  that many HW based implementations may have one set
+ * of limits for the HW and also support a SW TM implementation with a
+ * (presumably larger) different set of limits.  There are also cases where
+ * there could be more than SW implementation (one supporting say tens of
+ * thousands of tm_queues and a variant supporting tens of millions of
+ * tm_queues). This is needed as platforms might have different constraints
+ * for different egress type and there is no way to know them using
+ * odp_tm_capabilities() provided capabilities. The caller passes in an array
+ * of odp_tm_capabilities_t records  and the number of such records.  Then the
+ * first N of these records will be  filled in by the implementation and the
+ * number N will be returned.  In the event that N is larger than the
+ * capabilities_size, N will still be returned, but only capabilities_size
+ * records will be filled in.
```

Comment:
> Then the first N of these records will be  filled in by the implementation and the number N will be returned.  In the event that N is larger than the capabilities_size, N will still be returned, but only capabilities_size records will be filled in.

This is not quite right and needs to be fixed (N cannot be both the total number of capabilities and the number of records filled in the array). See e.g. odp_crypto_cipher_capability() for alternate wording.

Reply:
This has not been fixed. The quoted text still appears unchanged in the code.

Reply:
I changed it to for new API. Do you mean for existing API ?
". If number of such records implementation supports is larger than
 * caller requested number, then only caller requested number of records are
 * written and return value is max number of records implementation supports.
 * Caller then may again call with larger number of records to be returned."

Reply:
I got confused and looked at odp_tm_capabilities() which contains the above text and which this patch set has not changes. As that too is a bit inaccurate, it could also be fixed.

Reply:
Ack.